### DISCUSSION TITLE
Why are logical timestamps only partially orderable?

### DISCUSSION THREAD

**Discussion by sgpinkus** [streamich/json-joy#765] [Q&A]
The [Logical Clock](https://jsonjoy.com/specs/json-crdt-patch/patch-document/logical-clock) doc states:

> A logical timestamp is a globally unique identifier, which can be partially ordered. .. 

Yet AFAICT the comparison procedure described on this page gives a total ordering. Also the LWW implementation seems to rely on total ordering. Should this say  "A logical timestamp is a globally unique identifier, which can be *totally* ordered. .."?

**Comment by marolbrich** [comment]
Logical clocks initially only define a partial order. If the session IDs are also used for ordering, it is totally ordered. I think both statements are correct.

**Comment by sgpinkus** [comment]
The session ID is defined as part of the logical clock.

> A logical timestamp is a globally unique identifier, which can be partially ordered. A logical timestamp is a tuple of two values: (1) a session ID; and (2) a sequence counter.

**Comment by marolbrich** [comment]
You are right. However, the text is correct as well. I guess, the author wanted to stress the point that the causal order defined by the sequence counter defines only a partial order. The use of the session ID to force a total order is necessary but not justified by causality.
